coffeecard/CoffeeCard.WebApi/Controllers/v2/AccountController.cs

```diff
@@ -220,9 +224,47 @@ private async Task<UserResponse> UserWithRanking(User user)
         [ProducesResponseType(typeof(ApiError), StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(typeof(UserSearchResponse), StatusCodes.Status200OK)]
         [Route("search")]
-        public async Task<ActionResult<UserSearchResponse>> SearchUsers([FromQuery][Range(0, int.MaxValue)] int pageNum, [FromQuery] string filter = "", [FromQuery][Range(1, 100)] int pageLength = 30)
+        public async Task<ActionResult<UserSearchResponse>> SearchUsers(
+            [FromQuery][Range(0, int.MaxValue)] int pageNum, [FromQuery] string filter = "",
+            [FromQuery][Range(1, 100)] int pageLength = 30)
         {
             return Ok(await _accountService.SearchUsers(filter, pageNum, pageLength));
         }
+
+        /// <summary>
+        /// Sends a magic link to the user's email to login
+        /// </summary>
+        /// <param name="request">User's email</param>
+        /// <returns></returns> 
+        [HttpPost]
+        [AllowAnonymous]
+        [Route("login")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesDefaultResponseType]
+        public async Task<ActionResult> Login([FromBody] UserLoginRequest request)
+        {
+            await _accountService.SendMagicLinkEmail(request.Email, request.LoginType);
+            return new NoContentResult();
+        }
+
+        /// <summary>
+        /// Authenticates the user with the token hash from a magic link
+        /// </summary>
+        /// <param name="tokenHash">The token hash from the magic link</param>
+        /// <returns>A JSON Web Token used to authenticate for other endpoints and a refresh token to re-authenticate without a new magic link</returns>
+        [HttpPost]
+        [AllowAnonymous]
+        [Route("auth")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(MessageResponseDto), StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<ActionResult<UserLoginResponse>> Authenticate(string tokenHash)
```

Comment:
In case we would like to add properties to the request, I'd prefer the tokenhash be wrapped in a request object

Reply:
Done in https://github.com/AnalogIO/analog-core/pull/289/commits/d5c5e1203d6fa43717b2509cff10cbeff747be13